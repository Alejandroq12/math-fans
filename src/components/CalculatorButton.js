// eslint-disable-next-line import/no-extraneous-dependencies
import PropTypes from 'prop-types';

function CalculatorButton({ value, operation }) {
  return (
    <button
      type="button"
      className={`calculator-button calculator-button-${value} calculator-button${
        operation ? ' operation' : ''
      }`}
    >
      {value}
    </button>
  );
}

CalculatorButton.propTypes = {
  value: PropTypes.string.isRequired,
  operation: PropTypes.bool,
};

CalculatorButton.defaultProps = {
  operation: false,
};

export default CalculatorButton;
