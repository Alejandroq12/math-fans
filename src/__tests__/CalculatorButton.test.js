import { render, fireEvent } from '@testing-library/react';
import CalculatorButton from '../components/CalculatorButton';

describe('CalculatorButton', () => {
  it('renders the calculator button component', () => {
    const onClick = jest.fn();
    const { asFragment } = render(<CalculatorButton value="1" onClick={onClick} />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('should call onClick when button is clicked', () => {
    const onClick = jest.fn();
    const { getByText } = render(<CalculatorButton value="1" onClick={onClick} />);
    fireEvent.click(getByText('1'));
    expect(onClick).toHaveBeenCalledTimes(1);
  });
});
