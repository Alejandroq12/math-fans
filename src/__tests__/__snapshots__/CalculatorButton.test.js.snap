// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CalculatorButton renders the calculator button component 1`] = `
<DocumentFragment>
  <button
    class="calculator-button calculator-button-1 calculator-button"
    type="button"
  >
    1
  </button>
</DocumentFragment>
`;
